### PR TITLE
API Security RC tests

### PR DESCRIPTION
**Pull request by uurien**
## Motivation
Have tests to check that we are applying Api Security related RC correctly

## Changes

- Add new scenario for these tests 
- Add the new scenario to scenarios to run by the tracer
- Write some documentation about creating new RC data
- New RC data source of RC data: https://github.com/DataDog/rc-tracer-client-test-generator/pull/8/files
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [x] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/_context/_scenarios.py [hunk 1]
```diff
@@ -1266,6 +1266,19 @@ class scenarios:
         """,
     )
 
+    appsec_api_security_rc = EndToEndScenario(
```
Comment (cbeauchesne) [inline utils/_context/_scenarios.py:1269]: Can you add this new scenario in https://github.com/DataDog/system-tests/blob/main/.github/workflows/ci.yml ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Comment by cbeauchesne** [comment]
Updating the branch, some issue has been fixed in other PR

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
Approved the framework usage.

It may be useful to request a review from somebody that knows the feature, to review the test logic.

File: tests/appsec/api_security/test_api_security_rc.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+from utils import (
+    context,
+    coverage,
```
Comment (cbeauchesne) [inline tests/appsec/api_security/test_api_security_rc.py:7]: I've just removed this in #2091, you can remove the import, and the later references

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/appsec/api_security/test_api_security_rc.py (hunk 2 above)
Comment (christophe-papazian) [reply tests/appsec/api_security/test_api_security_rc.py:7]: Thank you for the fix !

Comment (cbeauchesne) [reply tests/appsec/api_security/test_api_security_rc.py:7]: (sorry for the trouble 😓 )

**Review by christophe-papazian — APPROVED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**